src/chains/ethereum/ethereum/src/blockchain.ts

```diff
@@ -1324,67 +1356,15 @@ export default class Blockchain extends Emittery<BlockchainTypedEvents> {
       }
     };
 
-    // Don't even let the vm try to flush the block's _cache to the stateTrie.
-    // When forking some of the data that the traced function may request will
-    // exist only on the main chain. Because we pretty much lie to the VM by
-    // telling it we DO have data in our Trie, when we really don't, it gets
-    // lost during the commit phase when it traverses the "borrowed" datum's
-    // trie (as it may not have a valid root). Because this is a trace, and we
-    // don't need to commit the data, duck punching the `flush` method (the
-    // simplest method I could find) is fine.
-    // Remove this and you may see the infamous
-    // `Uncaught TypeError: Cannot read property 'pop' of undefined` error!
-    (vm.stateManager as any)._cache.flush = () => {};
-
-    // Process the block without committing the data.
-    // The vmerr key on the result appears to be removed.
-    // The previous implementation had specific error handling.
-    // It's possible we've removed handling specific cases in this implementation.
-    // e.g., the previous incantation of RuntimeError
-    await vm.stateManager.checkpoint();
-    try {
-      for (let i = 0, l = newBlock.transactions.length; i < l; i++) {
-        const tx = newBlock.transactions[i] as any;
-        if (tx === transaction) {
-          if (keys && contractAddress) {
-            const database = this.#database;
-            const ejsContractAddress = { buf: contractAddress } as any;
-            await Promise.all(
-              keys.map(async key => {
-                // get the raw key using the hashed key
-                const rawKey = await database.storageKeys.get(key);
-
-                const result = await vm.stateManager.getContractStorage(
-                  ejsContractAddress,
-                  rawKey
-                );
-
-                storage[Data.toString(key, key.length)] = {
-                  key: Data.from(rawKey, rawKey.length),
-                  value: Data.from(result, 32)
-                };
-              })
-            );
-            break;
-          } else {
-            vm.on("step", stepListener);
-            // force the loop to break after running this transaction by setting
-            // the current iteration past the end
-            i = l;
-          }
-        }
-        this.emit("ganache:vm:tx:before", {
-          context: transactionEventContext
-        });
-        await vm.runTx({ tx, block: newBlock as any });
-        this.emit("ganache:vm:tx:after", {
-          context: transactionEventContext
-        });
-      }
-      vm.removeListener("step", stepListener);
-    } finally {
-      await vm.stateManager.revert();
-    }
+    vm.on("step", stepListener);
+    this.emit("ganache:vm:tx:before", {
+      context: transactionEventContext
+    });
+    await vm.runTx({ tx: transaction as any, block: newBlock as any });
+    this.emit("ganache:vm:tx:after", {
+      context: transactionEventContext
+    });
```

Comment:
This part needed to be added here, as it was previously handled inside the weird loop that had the condition specifically for `storageRangeAt`, here:  https://github.com/trufflesuite/ganache/pull/3470/files#diff-eb13881c7bcdaf19f3b7fe3f7c0b32570442528d3511e250621734dbbcb74992L1379